control/grpc.py

```diff
@@ -1741,12 +1741,27 @@ def get_subsystem_ha_status(self, nqn) -> bool:
     def matching_listener_exists(self, context, nqn, gw_name, traddr, trsvcid) -> bool:
         if not context:
             return False
-        listener_key = GatewayState.build_listener_key(nqn, gw_name, "TCP", traddr, trsvcid)
+
+        key_prefix = ""
+        key_suffix = ""
         state = self.gateway_state.local.get_state()
-        if state.get(listener_key):
+        # We want to check for all the listeners for this address and port, regardless of the gateway
+        listener_key = GatewayState.build_listener_key(nqn, "", "TCP", traddr, trsvcid)
+        try:
+            key_prefix, key_suffix = listener_key.split("__")
```

Comment:
Can't we use a constant instead of just "__" ?

Reply:
@caroav any constant used there should be something which can't be used in a gateway name.